projects/igniteui-angular/src/lib/directives/mask/mask-parsing.service.ts

```diff
@@ -156,8 +157,8 @@ export class MaskParsingService {
         }
 
         if (value.length <= 1) {
-            let isDelete = false;
```

Comment:
Let's try to simplify this code a little bit. We can get rid of this whole check (lines 159 through 179) and then use the above loop to remove characters step by step as soon as a `delete` key is pressed. To do that, we can check if the `value` that we've received has any length to it, since on a `delete` it should be empty (even if we have some text selection beforehand). Then we need to make sure that we are not on a mask literal, since we don't want these deleted and finally if both of these conditions are met we can set the `maskedValue` to be a `promptChar`, which we can take from the `maskOptions`. In all cases we, should update the `cursor`, to make sure that it's always at the end of the selection/that it goes forward when `delete` is pressed.